app/models/concerns/hyku_addons/account_behavior.rb

```diff
@@ -3,9 +3,14 @@
 module HykuAddons
   module AccountBehavior
     extend ActiveSupport::Concern
-
     included do
       belongs_to :datacite_endpoint, dependent: :delete
+      has_many :children, class_name: "Account", foreign_key: "parent_id", dependent: :destroy, inverse_of: :parent
+      belongs_to :parent, class_name: "Account", inverse_of: :parent, foreign_key: "parent_id", optional: true
+
+      store_accessor :data, :is_parent
```

Comment:
This is moving things just as they are in the account file in the main app to the rails engine, since when we start moving works we can  check if they are used and where. If they are in the file they were used at some point  that is my thinking

Reply:
That makes sense.  Are the parent/children associations for the shared index/search?  I'm wondering if we really still need them.

Reply:
Originally yes, but it is also to be  used to show the owner of a group of tenants eg BL is the tenant that brought Mola, NMS, Kew etc.

Reply:
Cool, does this show up in the interface anywhere?  Since this is a UP thing that's new to me I'm curious how it gets used.

Reply:
It is only used for shared search currently but even after you change to the suggested new implementation, this will be useful in settings. I see it being used in  settings, to allow a main tenant like BL etc, to administer aspects of a child tenant settings not available in the super admin settings page.

Secondly, when Brian starts signing the various consortium by that I mean  group of journals or group of libraries that he has been talking to, there will be need to group them in some way, based on the lead tenant.

Do you want it removed?, I mean parent/children associations

Reply:
I'm on the fence about this.  On the one hand it would be nice to cut complexity and only bring it in when it is needed, but on the other hand it will make migrations harder if we drop it.  So let's keep it for now to not slow the momentum and we can worry about it later if it proves to be an issue.